frame/nomination-pools/src/lib.rs

```diff
@@ -1194,18 +1170,26 @@ pub mod pallet {
 			// Unbond in the actual underlying pool
 			T::StakingInterface::unbond(bonded_pool.bonded_account(), balance_to_unbond)?;
 
-			// Note that we lazily create the unbonding pools here if they don't already exist
-			let sub_pools = SubPoolsStorage::<T>::get(delegator.pool_id).unwrap_or_default();
 			let current_era = T::StakingInterface::current_era();
 			let unbond_era = T::StakingInterface::bonding_duration().saturating_add(current_era);
-
-			// Merge any older pools into the general, era agnostic unbond pool. Note that we do
-			// this before inserting to ensure we don't go over the max unbonding pools.
-			let mut sub_pools = sub_pools.maybe_merge_pools(unbond_era);
+			// Note that we lazily create the unbonding pools here if they don't already exist
+			let mut sub_pools = SubPoolsStorage::<T>::get(delegator.pool_id)
+				.unwrap_or_default()
+				.maybe_merge_pools(unbond_era);
 
 			// Update the unbond pool associated with the current era with the unbonded funds. Note
 			// that we lazily create the unbond pool if it does not yet exist.
-			sub_pools.unchecked_with_era_get_or_make(unbond_era).issue(balance_to_unbond);
+			if !sub_pools.with_era.contains_key(&unbond_era) {
+				sub_pools
+					.with_era
+					.try_insert(unbond_era, UnbondPool::default())
+					.expect("caller has checked pre-conditions. qed."); // TODO: i do NOT see any proof for this.
```

Comment:
```suggestion
                     // This should never fail because the above call to `maybe_merge_pools` should
                     // ensure there is always enough space to insert. But since this call is transactional
                     // we defensively return an error
					.try_insert(unbond_era, UnbondPool::default())?;
```